### PR TITLE
add doc on proof ... qed|admit|abort + remove sections subdir

### PR DESCRIPTION
**Pull request by fblanqui**

### CODE REVIEW

**Comment by fblanqui** [comment]
The new doc can be visualized on https://github.com/fblanqui/lambdapi/blob/docdir/doc/DOCUMENTATION.md.

File: doc/DOCUMENTATION.md [hunk 1]
```diff
@@ -9,36 +9,34 @@ project.
...
- - [Syntax of terms](sections/terms.md)
+ - [Syntax of terms](terms.md)
 
- - [Commands](sections/commands.md)
+ - [Commands](commands.md)
 
- - [Tactics](sections/tactics.md)
+ - [Tactics](tactics.md)
 
- - [Editing Lambdapi source code with Emacs](sections/emacs.md)
```
Comment (gabrielhdt) [inline doc/DOCUMENTATION.md:30]: There is no more entry for emacs, is it wanted?

File: doc/terms.md [hunk 2]
```diff
@@ -1,13 +1,25 @@
...
 file `dir1/` ... `/dirn/file.lp`. To be used, `dir1.` ... `dirn.file`
 must be required first.
 
+An identifier can be:
+
+  * an ASCII sequence of characters in the regular language `[a-zA-Z_][a-zA-Z0-9_]*`
+
+  * a non-negative integer if the builtins "0" and "+1" have been previously set (see the [command](commands.md) `set builtin`)
+
+  * a unicode x character previously declared using the [command](commands.md) `set declared "x"`
```
Comment (gabrielhdt) [inline doc/terms.md:17]: I would more simply say
> a unicode character that has been declared using the ...

File: doc/DOCUMENTATION.md (hunk 1 above)
Comment (fblanqui) [reply doc/DOCUMENTATION.md:30]: Yes, it is moved into ui.md.

**Comment by fblanqui** [comment]
@gabrielhdt I took into account your remarks. Thanks. Do you have more?

File: doc/DOCUMENTATION.md (hunk 1 above)
Comment (gabrielhdt) [reply doc/DOCUMENTATION.md:30]: So personally, I'd prefer to keep a slightly longer table of content (it's not that long to read...) to have more accessible pages and less "click paths" to reach a page.

**Comment by gabrielhdt** [comment]
> @gabrielhdt I took into account your remarks. Thanks. Do you have more?

I made one in the above conversation.

File: doc/DOCUMENTATION.md (hunk 1 above)
Comment (fblanqui) [reply doc/DOCUMENTATION.md:30]: Good point. I gave more structure to the toc. Is it better now?

Comment (gabrielhdt) [reply doc/DOCUMENTATION.md:30]: Yes, that's nice

**Pull request merged**